test/e2e/apimachinery/aggregator.go

```diff
@@ -570,3 +578,53 @@ func generateFlunderName(base string) string {
 	}
 	return fmt.Sprintf("%s-%d", base, id)
 }
+
+func deleteAPIService(f *framework.Framework, aggrclient *aggregatorclient.Clientset) {
+	// This function allows for clean execution when having HA control planes. We need to allow time for
+	// the sample api server to be unregistered from all api servers before tearing down the deployment.
+	framework.Logf("Waiting on v1alpha1.wardle.example.com to disappear from all apiserver instances")
+	client := f.ClientSet
+
+	err := aggrclient.ApiregistrationV1().APIServices().Delete(context.TODO(), "v1alpha1.wardle.example.com", metav1.DeleteOptions{})
+	framework.ExpectNoError(err, "deleting apiservice %s", "v1alpha1.wardle.example.com")
+
+	endpoints, err := client.CoreV1().Endpoints("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+	framework.ExpectNoError(err, "getting kubernetes endpoints in namespace default")
+	var apiPort int32 = 0
+	apiAddresses := make([]string, 0)
+	for _, subset := range endpoints.Subsets {
+		for _, port := range subset.Ports {
+			if port.Name != "https" || port.Protocol != "TCP" || port.Port == 0 {
+				continue
+			}
+			apiPort = port.Port
+		}
+		for _, address := range subset.Addresses {
+			apiAddresses = append(apiAddresses, address.IP)
+		}
+	}
+	if len(apiAddresses) == 0 || apiPort == 0 {
+		framework.Failf("failed to get apiserver ip addresses or port")
+	}
+	execPod := e2epod.CreateExecPodOrFail(f.ClientSet, f.Namespace.Name, "verify-apiserver-response", nil)
+	err = wait.PollImmediate(periodAPIDelete, timeoutAPIDelete, func() (bool, error) {
+		for _, address := range apiAddresses {
+			cmd := fmt.Sprintf(`curl -s -o /dev/null -w '%%{http_code}' https://kubernetes:%d/apis/wardle.example.com/v1alpha1 --resolve kubernetes:%d:%s --cacert %s -H "Authorization: Bearer $(cat %s)"`,
```

Comment:
We don't need globbing here since the ipv6 happens outside the URL. Anyway I just included support for dual stack/IPv6 families. It should work even without brackets, but adding them for safety.